### PR TITLE
Microbenchmarks the Transformer block.

### PR DESCRIPTION
**Pull request by wujingyue**
For #2199.

I've been maintaining the nvFusions and the inputs in a branch. This PR checks them into nvFuser's main for convenience.

```shell
$ pytest benchmarks/python/test_transformer.py
================================================================================================ test session starts =================================================================================================
platform linux -- Python 3.10.12, pytest-8.1.1, pluggy-1.5.0
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /opt/pytorch/nvfuser
plugins: xdist-3.6.1, hypothesis-6.104.2, timestamper-0.0.10, cov-5.0.0, timeout-2.3.1, random-order-1.1.1, benchmark-4.0.0, shard-0.1.2
collected 2 items
Running 2 items in this shard

benchmarks/python/test_transformer.py ..                                                                                                                                                                       [100%]

--------------------------------------------------------------------------------------- benchmark: 2 tests --------------------------------------------------------------------------------------
Name (time in ms)                  Min                 Max                Mean            StdDev              Median               IQR            Outliers      OPS            Rounds  Iterations
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_transformer_forward       54.7508 (1.0)       72.9630 (1.0)       67.7275 (1.0)      4.8153 (1.38)      68.7300 (1.0)      2.0469 (1.01)          2;2  14.7650 (1.0)          10           1
test_transformer_backward     174.6965 (3.19)     187.7991 (2.57)     183.9202 (2.72)     3.4975 (1.0)      184.6459 (2.69)     2.0344 (1.0)           2;1   5.4371 (0.37)         10           1
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
================================================================================================= 2 passed in 8.56s ==================================================================================================
```

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

File: benchmarks/python/test_transformer.py [hunk 1]
```diff
@@ -0,0 +1,1250 @@
...
+#              )
+# +            print(self.getReproErrorString("executing", inputs))
+#          except Exception as err:
+# -            logger.exception(self.getReproErrorString("executing", inputs))
+#              raise
+#
+#          return result
+# ```
+#
+# 5. `pytest thunder/benchmarks/targets.py -k 'nanogpt_block and backward-thunder]' -s`
```
Comment (cowanmeg) [inline benchmarks/python/test_transformer.py:35]: missing a [
-k nanogpt_block[backward-thunder]

**Review by cowanmeg — APPROVED** [review]

**Comment by wujingyue** [comment]
!build --pybench

**Pull request merged**